### PR TITLE
Add new options for command-line testing

### PR DESCRIPTION
**Pull request by ebogo1**
This is part of https://github.com/CesiumGS/cesium/issues/10134 - I think removing the standalone Jasmine SpecRunner files should be a follow-up PR so that this one focuses on the new command-line options for `npm run test`:

- `--debug`: overrides the `singleRun = true` flag we set in `karma.conf.cjs`, which allows users to use the "Debug" button in the Karma browser to use the browser debugger for unit tests
- `--includeName {Folder}/{FileName}`: does the same thing as clicking on a test suite in SpecRunner.html. This checks for string equality but could be changed to something else like `contains` if that makes more sense. I don't believe we have access to individual test names inside a suite, so the `fit` syntax is still needed to run specific tests alone.
- `--excludeName {Folder}/{FileName}`: same as `--includeName` but for excluding specs. This wasn't possible in SpecRunner.html but it mirrors the logic we have for including categories on the command line.

I also added a couple lines to the TestingGuide README. I recommend looking at the diff without whitespace changes.

### CODE REVIEW

File: Specs/customizeJasmine.js [hunk 1]
```diff
@@ -26,6 +28,12 @@ function customizeJasmine(
       return;
     }
 
+    if (includedName && name !== includedName) {
```
Comment (ggetz) [inline Specs/customizeJasmine.js:31]: Could we include partial matches? That way we can include all tests in a category. For example @ptrgags said he uses the web browser to run all `ModelExperimental` tests, which span multiple files.

File: Documentation/Contributors/TestingGuide/README.md [hunk 2]
```diff
@@ -197,6 +205,22 @@ It is also possible for Karma to run all tests against each browser installed on
...
 
+Alternatively, a single test suite can be run from the command line with the `includeName` option and a test name argument that matches the name in the `describe` function of the spec file. For example, the `Cartesian2` tests can be run with:
+
+`npm run test -- --includeName Core/Cartesian2`
+
+Similarly, test suites can be excluded with the `--excludeName` flag.
+
+#### Using Browser Debugging Tools
+
+If it is helpful to step through a unit test in a browser debugger, the `singleRun` Karma option can be disabled like so:
```
Comment (ggetz) [inline Documentation/Contributors/TestingGuide/README.md:216]:
```suggestion
If it is helpful to step through a unit test in a browser debugger, run the tests with the `debug` flag.
```

I don't think readers need or want to know the underlying karma confirmation we're using, so let's just omit it for simplicity's sake.

**Review by ggetz — COMMENTED** [review]
Thanks @ebogo1! I just have a few comments.

File: Specs/customizeJasmine.js (hunk 1 above)
Comment (ptrgags) [reply Specs/customizeJasmine.js:31]: So just for some more information on this, the SpecRunner query seems to do a case-sensitive substring search.

Though I think even a regex search would be easy to add with `new RegExp(query)` -- might be nice? then you could run custom queries like `--includeName 'Cesium3DTileset|ModelExperimental'`. Not necessary but might be nice to have

**Comment by ebogo1** [comment]
@ggetz I changed the string comparison to check for matching substrings in [b06c640](https://github.com/CesiumGS/cesium/pull/10170/commits/b06c6406eaf7c4f4d6824ebc099f3193c9a7e309) - should be good for another look. Per Peter's [comment](https://github.com/CesiumGS/cesium/pull/10170#discussion_r821742167) - I can add a regex check instead but I'm not sure how common that use case is. Yesterday we decided it might be overkill so I didn't make the change yet.

**Comment by ggetz** [comment]
> I can add a regex check instead but I'm not sure how common that use case is. Yesterday we decided it might be overkill so I didn't make the change yet.

I agree, we were looking to get to parody with the existing browser runner so this should be sufficient.

**Comment by ggetz** [comment]
Thanks @ebogo1! All the new options are working as expected by themselves and alongside other options.

**Pull request merged**